Feature/scout runs filecounts gsistats cylc8

amschne opened the pull request:
Updates sample workflow for use with cylc v8 and includes scripts for processing GSI analysis fit to observation equivalents for brightness temperatures as a function of satellite instrument channel. Changes include:

- updated workflow script ("flow.cylc") under "cylc8_test_flow/"
- changes to existing python scripts to import score_db as a module instead of using suprocess
- a new python script to call the GSI stats harvester for radiances and store associated metrics
- an "install_scripts.sh" utility to copy scripts/*.py into a workflow directory
- README updates

The cylc8_test_flow workflow has been tested via multiple applications against an atmosphere only scout run and a replay observer diagnostic, which archived GSI analysis "fit" files on AWS S3 buckets.

jrknezha commented:
Should this include a yaml for installing the appropriate conda package?

jrknezha commented on scripts/bucket_file_count.py:
```diff
@@ -52,13 +58,13 @@
 
-print("minutes: ")
-print(diff_minutes) 
+#print("minutes: ")
```
can we just remove the commented out print statements if we don't want to keep them?

jrknezha replied:
noting that I believe the use of having them previously was for understanding progress in the jobs when we look at logs. do we want to note any info for the logs?

jrknezha commented on test-scipy/flow.cylc:
```diff
@@ -0,0 +1,24 @@
+[scheduler]
```
what is the purpose of having this script? 

this doesn't seem necessary to the code base

jrknezha reviewed (commented):
Looking really good, a really major upgrade! Just a few comments to add on top of my previous question about having a yaml for the required packages to install for this version

amschne commented:
> Should this include a yaml for installing the appropriate conda package?

added a yaml, but not sure how to treat score-hv/db dependencies

amschne commented:
thanks @jrknezha - I cleaned up some of the scripts/*.py and added documentation in the README and in the test-scipy/flow.cylc script for the test-scipy workflow. Also, I added some logic to handle the yaml_file_path work directory in db_yaml_generator.py, which you may want to take a careful look at and/or test.

jrknezha reviewed (commented):
looking good! Tested the new change to the db-yaml-generator and it worked well. appreciate you addressing these comments just a couple last small things that should be quick and happy to approve from that

amschne commented on scripts/db_inc_logs.py:
```diff
@@ -39,47 +45,47 @@
 datetime_obj = dt.datetime.strptime(input_cycle, "%Y%m%dT%H")
-year = datetime_obj.strftime("%Y")
-month = datetime_obj.strftime("%m")
+#year = datetime_obj.strftime("%Y")
```
removed

amschne commented on README.md:
```diff
@@ -22,6 +22,18 @@ fail and retry.  After the file check is done, additional tasks will run such
+the correct importation and operation of numpy and scipy. This process runs 
+corresponding test suites on these libraries, ensuring the environment is 
+properly configured for scientific computing tasks.
+
```
added a sentence and link to cylc documentation

jrknezha reviewed (approved):
Excellent new functionality, thanks for addressing all my comments.
